### ISSUE TITLE
Have to implement Gradient Clipping in LinearRegression in Model.py 

### ISSUE THREAD

**Issue by ArkaStark** [ArkaStark/MachineLearning#3]
The Linear Regression model in Model.py is failing for certain learning rates due to the problem of vanishing/exploding gradients. This can be handled by a process called Gradient Clipping which keeps the gradients within a certain range.
